### PR TITLE
Warn when gate_headers are empty

### PR DESCRIPTION
**Pull request by isaacm**
Check for misconfigured gate_headers.

### CODE REVIEW

**Comment by TheSriram** [comment]
Could you add a test for this as well?

File: eom/bastion.py [hunk 1]
```diff
@@ -112,7 +112,11 @@ def wrap(app_backdoor, app_gated):
     # WSGI callable
     def middleware(env, start_response):
         path = env['PATH_INFO']
-        contains_gate_headers = set(gate_headers).issubset(set(env.keys()))
+        contains_gate_headers = (
+            set(gate_headers).issubset(set(env.keys())) and
+            len(gate_headers) > 0
+        )
+
```
Comment (BenjamenMeyer) [inline eom/bastion.py]: Perhaps a better solution is to wrap all of L114 to L121 (original) with the check value of L117; perhaps with a log warning that Bastion is in use but gate_headers is not configured.

Comment (isaacm) [inline eom/bastion.py]: See if my changes are what you had in mind @BenjamenMeyer .

File: eom/bastion.py [hunk 2]
```diff
@@ -111,14 +111,21 @@ def wrap(app_backdoor, app_gated):
...
+                    if not contains_gate_headers:
+                        return app_backdoor(env, start_response)
+                    else:
+                        return _http_gate_failure(start_response)
+        else:
+            LOG.warn(
+                "Bastion is in use and gate_headers option is not configured "
+                "or it is set to allow all requests to unrestricted route(s)."
+            )
+            return app_backdoor(env, start_response)
```
Comment (BenjamenMeyer) [inline eom/bastion.py]: Remove L1228 and let it fall through to L131. L126 is also incorrect.

It should never default to be unrestricted.

**Comment by isaacm** [comment]
I addressed the review items a while back and retested using tox today all the tests pass.
Were there any other items of concern?

**Comment by BenjamenMeyer** [comment]
:+1:

**Comment by TheSriram** [comment]
:+1:

**Pull request merged**